plugin/discovery/version_set.go

```diff
@@ -0,0 +1,58 @@
+package discovery
+
+import (
+	version "github.com/hashicorp/go-version"
+)
+
+// A ConstraintsStr is a string containing a possibly-invalid representation
+// of a version constraint provided in configuration. Call Parse on it to
+// obtain a real Constraint object, or discover that it is invalid.
+type ConstraintsStr string
```

Comment:
Could this be singular `ContstraintStr`? (I like to avoid adding pascal/hungarian types where possible too, but I see this matches `VersionStr` which is separate from `Version`)